pkg/controller/kms/key/setup.go

```diff
@@ -0,0 +1,336 @@
+	client svcsdkapi.KMSAPI
+}
+
+func (u *updater) update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) {
+	cr, ok := mg.(*svcapitypes.Key)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errUnexpectedObject)
+	}
+
+	if cr.Spec.ForProvider.Description != nil {
+		if _, err := u.client.UpdateKeyDescriptionWithContext(ctx, &svcsdk.UpdateKeyDescriptionInput{
+			KeyId:       aws.String(meta.GetExternalName(cr)),
+			Description: cr.Spec.ForProvider.Description,
+		}); err != nil {
+			return managed.ExternalUpdate{}, errors.Wrap(err, errUpdate)
+		}
+	}
+
+	// Policy
+	if _, err := u.client.PutKeyPolicyWithContext(ctx, &svcsdk.PutKeyPolicyInput{
+		KeyId:      aws.String(meta.GetExternalName(cr)),
+		PolicyName: aws.String("default"),
+		Policy:     cr.Spec.ForProvider.Policy,
+	}); err != nil {
+		return managed.ExternalUpdate{}, errors.Wrap(err, errUpdate)
+	}
+
+	// Tags
+	if err := u.updateTags(ctx, cr); err != nil {
+		return managed.ExternalUpdate{}, err
+	}
+
+	// Enable / Disable
+	if err := u.enableDisableKey(ctx, cr); err != nil {
+		return managed.ExternalUpdate{}, err
+	}
+
+	return managed.ExternalUpdate{}, nil
+}
+
+func (u *updater) updateTags(ctx context.Context, cr *svcapitypes.Key) error {
+	tagsOutput, err := u.client.ListResourceTagsWithContext(ctx, &svcsdk.ListResourceTagsInput{
+		KeyId: aws.String(meta.GetExternalName(cr)),
+	})
+	if err != nil {
+		return errors.Wrap(err, errUpdate)
+	}
+
+	addTags, removeTags := diffTags(cr.Spec.ForProvider.Tags, tagsOutput.Tags)
+
+	if len(addTags) != 0 {
+		if _, err := u.client.TagResourceWithContext(ctx, &svcsdk.TagResourceInput{
+			KeyId: aws.String(meta.GetExternalName(cr)),
+			Tags:  addTags,
+		}); err != nil {
+			return err
+		}
+	}
+	if len(removeTags) != 0 {
+		if _, err := u.client.UntagResourceWithContext(ctx, &svcsdk.UntagResourceInput{
+			KeyId:   aws.String(meta.GetExternalName(cr)),
+			TagKeys: removeTags,
+		}); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func isUpToDateEnableDisable(cr *svcapitypes.Key) bool {
+	return aws.BoolValue(cr.Spec.ForProvider.Enabled) == aws.BoolValue(cr.Status.AtProvider.Enabled)
+}
+
+func (u *updater) enableDisableKey(ctx context.Context, cr *svcapitypes.Key) error {
+	if isUpToDateEnableDisable(cr) {
+		return nil
+	}
+
+	if aws.BoolValue(cr.Spec.ForProvider.Enabled) {
+		if _, err := u.client.EnableKeyWithContext(ctx, &svcsdk.EnableKeyInput{
+			KeyId: aws.String(meta.GetExternalName(cr)),
+		}); err != nil {
+			return err
+		}
+	} else {
+		if _, err := u.client.DisableKeyWithContext(ctx, &svcsdk.DisableKeyInput{
+			KeyId: aws.String(meta.GetExternalName(cr)),
+		}); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+type deleter struct {
+	client svcsdkapi.KMSAPI
+}
+
+// schedule for deletion instead of delete
+func (d *deleter) delete(ctx context.Context, mg resource.Managed) error {
+	cr, ok := mg.(*svcapitypes.Key)
+	if !ok {
+		return errors.New(errUnexpectedObject)
+	}
+
```

Comment:
```suggestion
		cr.SetConditions(xpv1.Deleting())
```

we usually set condition here to make sure we notify user that we're trying to delete even if the deletion call results in error.